screens/SnoozeScreen.js

```diff
@@ -1,41 +1,109 @@
 import React, { Component } from "react";
-import {
-  StyleSheet,
-  Button
-} from "react-native";
+import { StyleSheet, Alert } from "react-native";
+import { Button } from "../components/buttons";
 import { Actions } from "react-native-router-flux";
 import { Container, Content, Header, View } from "../components/layout";
 import { Text } from "../components/typography";
+import { connect } from 'react-redux';
+import { increaseTime, decreaseTime, countdown, clearTime, resetTime } from '../store/actions';
 
-const SnoozeScreen = () => {
-  return (
-    <Container>
-    <Header>Snooze Mode</Header>
-    <Content>
-      <View style={styles.container}>
-        <Text style={styles.welcome}>
-          Snooze Screen
-        </Text>
-        <Button title="Snooze" onPress={() => Actions.alarm()} />
-      </View>
-    </Content>
-    </Container>
-  );
+class SnoozeScreen extends Component {
+  constructor(props) {
+    super(props);
+    this.countdown = this.countdown.bind(this);
+    this.snoozeHandler = this.snoozeHandler.bind(this);
+  }
+
+  countdown() {
+    console.log(this.props.time);
+    if (this.props.time.timeRemaining - 1 <= 0) {
+      this.props.clearTime();
+      clearInterval(this.interval);
+      Alert.alert("Help request sent", "Help request has been sent to your responder network", [{text: 'OK', onPress: () => Actions.main()}], {cancelable: false});
+    } else {
+      this.props.countdown(this.props.time.timeRemaining);
+    }
+  };
+
+  convertSeconds = (seconds) => {
```

Comment:
Would likely want to make these into utility functions that can be used throughout app. I think I would maybe consider that we start using "momentjs" for time and date manipulations and stuff, but maybe a bit overkill too

Reply:
You are right. This should be extracted into a service class to deal with time related conversions and calculations. However, I wouldnt add a dependency just for this. 

Creating the service and refactoring will be another task.